creator-node/src/routes/files.js

```diff
@@ -247,46 +252,89 @@ module.exports = function (app) {
     // Do not act as a public gateway. Only serve IPFS files that are tracked by this creator node.
     const dirCID = req.params.dirCID
     const filename = req.params.filename
+    const ipfsPath = `${dirCID}/${filename}`
 
-    const queryResults = await models.File.findOne({ where: {
-      multihash: dirCID,
-      type: 'dir'
-    } })
+    const queryResults = await models.File.findOne({
+      where: {
+        storagePath: { [models.Sequelize.Op.like]: `%${dirCID}%` },
+        // Could look like blob/150x150.jpg or artwork/150x150.jpg
+        sourceFile: { [models.Sequelize.Op.like]: `%${filename}` },
+        type: { [models.Sequelize.Op.ne]: 'dir' }
+      }
+    })
     if (!queryResults) {
       return sendResponse(req, res, errorResponseNotFound(`No dir entry found for provided dirCID: ${dirCID}`))
     }
 
-    // Conditionally re-add from filestorage to IPFS
-    try {
-      await rehydrateIpfsFromFsIfNecessary(
+    redisClient.incr('ipfsStandaloneReqs')
+    const totalStandaloneIpfsReqs = parseInt(await redisClient.get('ipfsStandaloneReqs'))
+    logger.info(`IPFS Standalone Request - ${ipfsPath}`)
+    logger.info(`IPFS Stats - Standalone Requests: ${totalStandaloneIpfsReqs}`)
+
+    if (req.query.fromFS && req.query.fromFS === 'true') {
+      logger.info(`Retrieving ${ipfsPath} directly from filesystem`)
+      // Retrieves the file directly from the filesystem rather than checking IPFS first
+
+      // Fire-and-forget rehydration
+      rehydrateIpfsFromFsIfNecessary(
         req,
         dirCID,
         queryResults.storagePath,
         filename
       )
-    } catch (e) {
-      // If rehydrate throws error, return 500 without attempting to stream file.
-      return sendResponse(req, res, errorResponseServerError(e.message))
-    }
 
-    // TODO - check if file with filename is also stored in CNODE
+      return streamFromFileSystem(req, res, queryResults.storagePath)
+    } else {
+      // Retrieves the file from IPFS and falls back to the filesystem if unavailable
+      logger.info(`Attempting to retrieve ${ipfsPath} from IPFS`)
 
-    const ipfsPath = `${dirCID}/${filename}`
+      // Conditionally re-add from filestorage to IPFS
+      try {
+        await rehydrateIpfsFromFsIfNecessary(
+          req,
+          dirCID,
+          queryResults.storagePath,
+          filename
+        )
+      } catch (e) {
+        // If rehydrate throws error, return 500 without attempting to stream file.
+        return sendResponse(req, res, errorResponseServerError(e.message))
+      }
 
-    redisClient.incr('ipfsStandaloneReqs')
-    const totalStandaloneIpfsReqs = parseInt(await redisClient.get('ipfsStandaloneReqs'))
-    logger.info(`IPFS Standalone Request - ${ipfsPath}`)
-    logger.info(`IPFS Stats - Standalone Requests: ${totalStandaloneIpfsReqs}`)
+      try {
+        await new Promise((resolve, reject) => {
```

Comment:
separate but this catReadableStream call is made in both ipfs routes and could be moved to its own function